pkg/skaffold/build/cluster/secret.go

```diff
@@ -24,62 +24,69 @@ import (
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes"
 )
 
+const (
+	defaultKanikoSecretPath = "kaniko-secret"
+)
+
 func (b *Builder) setupPullSecret(out io.Writer) (func(), error) {
 	if b.PullSecret == "" && b.PullSecretName == "" {
 		return func() {}, nil
 	}
 
-	color.Default.Fprintf(out, "Creating kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
-
+	color.Default.Fprintf(out, "Checking for kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
 	client, err := kubernetes.Client()
 	if err != nil {
 		return nil, fmt.Errorf("getting Kubernetes client: %w", err)
 	}
 
 	secrets := client.CoreV1().Secrets(b.Namespace)
-
-	if b.PullSecret == "" {
-		logrus.Debug("No pull secret specified. Checking for one in the cluster.")
-
-		if _, err := secrets.Get(b.PullSecretName, metav1.GetOptions{}); err != nil {
-			return nil, fmt.Errorf("checking for existing kaniko secret: %w", err)
+	if _, err := secrets.Get(b.PullSecretName, metav1.GetOptions{}); err != nil {
+		color.Default.Fprintf(out, "Creating kaniko secret [%s/%s]...\n", b.Namespace, b.PullSecretName)
+		if b.PullSecret == "" {
+			return nil, fmt.Errorf("secret %s does not exisit. No path specified to create it", b.PullSecretName)
 		}
-
+		return createSecretFromFile(secrets, b.PullSecretName, b.PullSecret)
+	}
+	if b.PullSecret == "" {
+		// TODO: Remove the warning when pod health check can display pod failure errors.
+		logrus.Warnf("Setting secret keyfile path to %s. If this is incorrect, please specify using config key `pullSecret`.\nSee https://skaffold.dev/docs/references/yaml/#build-cluster-pullSecret", defaultKanikoSecretPath)
```

Comment:
do we really want a warning here? I think a debug level is sufficient, as the user should only see this when there is something wrong. For those users who have 100% right with the default value, this just adds extra noise.

Reply:
it does add to extra noise. However, it is not documented very well that skaffold expects the secret to be at path `kaniko-secret` when creating a secret. 
Untill, we can provide an actionable error, i feel the noise is fine.